File: app/src/main/java/com/codelixir/ioioservo/NotificationActionBroadcastReceiver.java
package com.codelixir.ioioservo;

import android.app.NotificationManager;
import android.bluetooth.BluetoothAdapter;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class NotificationActionBroadcastReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction() != null && intent.getAction().equals("stopService")) {
            context.stopService(new Intent(context, HelloIOIOService.class));
            context.startActivity(new Intent(context, MainActivity.class).setAction("stopActivity"));
        }
    }
}

File: app/src/main/java/com/codelixir/ioioservo/NotificationActionBroadcastReceiver.kt
package com.codelixir.ioioservo

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
import com.codelixir.ioioservo.HelloIOIOService

class NotificationActionBroadcastReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action != null && intent.action == "stopService") {
            context.stopService(Intent(context, HelloIOIOService::class.java))
            context.startActivity(
                Intent(
                    context,
                    MainActivity::class.java
                ).setAction("stopActivity").setFlags(FLAG_ACTIVITY_NEW_TASK)
            )
        }
    }
}